kythe/go/extractors/config/runextractor/mavencmd/pom_xml_modifier.go

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2018 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package mavencmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/beevik/etree"
+)
+
+// PreProcessPomXML takes a pom.xml file and either verifies that it already has
+// the bits necessary to specify a separate compiler on commandline, or adds
+// functionality by dropping in a maven-compiler-plugin to the build.
+//
+// Note this potentially modifies the input file, so make a copy beforehand if
+// you need to keep the original.
+func PreProcessPomXML(pomXMLFile string) error {
+	doc := etree.NewDocument()
+	err := doc.ReadFromFile(pomXMLFile)
+	if err != nil {
+		return fmt.Errorf("reading XML file %s: %v", pomXMLFile, err)
+	}
+	if hasCompilerPlugin(doc) {
+		return nil
+	}
+	if err := appendCompilerPlugin(doc); err != nil {
+		return err
+	}
+	f, err := os.OpenFile(pomXMLFile, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+	if err != nil {
+		return fmt.Errorf("opening file %s for append: %v", pomXMLFile, err)
```

Comment:
the errors from package `os` already include the path, this can be simpler:

    return fmt.Errorf("opening output: %v", err)

Reply:
Not actually using `O_APPEND` though?

Reply:
ah yea I copy/pasted this from a gradle where I was appending.  here I'm not, because the xml structure isn't amenable to it.

Reply:
I wonder if doc also includes path.  Guess we can clean that up later if it's spammy